dali/operators/generic/reshape.cc

```diff
@@ -247,10 +251,16 @@ void Reshape<Backend>::CalculateOutputShape(const Workspace &ws) {
   switch (shape_source_) {
     case ShapeSource::Arg:
       if (use_rel_shape_) {
-        output_shape_.resize(N, rel_uniform_shape_.size());
+        if (!src_dims_.empty()) {
+          DALI_ENFORCE(rel_uniform_shape_.size() == src_dims_.size(),
+            make_string(OpName(), ": ``src_dims`` and ``rel_shape`` should have the same length when both of them are provided."));
+        }
+    
+        output_shape_.resize(N, std::max(src_dims_.size(), rel_uniform_shape_.size()));
```

Comment:
```suggestion
        output_shape_.resize(N, rel_uniform_shape_.size());
```
max not needed, since you already checked the sizes match